experiment_buddy/experiment_buddy.py

```diff
@@ -234,17 +238,39 @@ def _setup_tb(logdir):
     return tensorboardX.SummaryWriter(logdir=logdir)
 
 
-def _ensure_scripts(extra_headers):
+def _ssh_session_connected():
+    """ TODO: move this to utils.py or to a class (better) """
+
+    kwargs_connection = {}
+
+    try:
+        kwargs_connection["host"] = os.environ["hostname"]
+    except KeyError:
+        raise EnvironmentError("Please add your hostname as env:" "\nhostname")
+
     try:
-        host = os.environ["MY_BUDDY_HOSTNAME"]
-        passwd = os.environ["MY_BUDDY_PASSWORD"]
+        kwargs_connection["connect_kwargs"] = {"password": os.environ["password"]}
     except KeyError:
-        raise EnvironmentError(
-            "Please add your server credentials as environment variables like so:"
-            "\nMY_BUDDY_HOSTNAME=hostname\nMY_BODY_PASSWORD=host password"
-        )
+        logging.warning("No ssh password given, if you did not ssh-copy-id, add it to the env as: MY_BUDDY_HOST_PASSWORD")
 
-    ssh_session = fabric.Connection(host=host, connect_kwargs={"password": passwd})
+    try:
+        kwargs_connection["port"] = os.environ["port"]
+    except KeyError:
+        logging.warning("No ssh port given, if you need it, add it to the env as: MY_BUDDY_HOST_PORT")
```

Comment:
You could move these warnings to the SSHException below, so that if everything works the user won't get many lines of messages